main/lsp/LSPMessage.cc

```diff
@@ -113,6 +113,23 @@ LSPMessage::LSPMessage(rapidjson::Document &d) : LSPMessage::LSPMessage(fromJSON
 
 LSPMessage::LSPMessage(const std::string &json) : LSPMessage::LSPMessage(fromJSON(json)) {}
 
+bool LSPMessage::isCanceled() const {
+    return canceled;
+}
+
+void LSPMessage::cancel() {
+    if (!canceled) {
+        canceled = true;
+        // Protect against nullptrs.
```

Comment:
Personal preference nit: although I'm generally in favor of more comments, this specific one didn't give me any extra information that wasn't clear to me from the code.  (I see that this code moved, and the original had this comment, so just giving you the liberty to remove if you found it equally tame.)

Reply:
Ah, I can add context: A Timer may not be specified (it's optional).